File: mobile/mami_driver/lib/features/rides/domain/models/payment_method.dart
enum RidePaymentMethod {
  cash('cash', 'Espèces'),
  airtelMoney('airtel_money', 'Airtel Money'),
  moovMoney('moov_money', 'Moov Money');

  const RidePaymentMethod(this.apiValue, this.label);

  final String apiValue;
  final String label;

  static RidePaymentMethod? fromApi(String? value) {
    if (value == null) return null;
    return RidePaymentMethod.values.firstWhere(
      (m) => m.apiValue == value,
      orElse: () => RidePaymentMethod.cash,
    );
  }
}

File: app/Modules/Municipality/Enums/PaymentMethod.php
<?php

namespace App\Modules\Municipality\Enums;

enum PaymentMethod: string
{
    case Cash = 'cash';
    case MobileMoney = 'mobile_money';
    case Card = 'card';
    case Transfer = 'transfer';

    public function label(): string
    {
        return match ($this) {
            self::Cash => 'Espèces',
            self::MobileMoney => 'Mobile Money',
            self::Card => 'Carte bancaire',
            self::Transfer => 'Virement',
        };
    }

    /**
     * @return list<string>
     */
    public static function values(): array
    {
        return array_column(self::cases(), 'value');
    }
}
